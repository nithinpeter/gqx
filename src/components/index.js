export * from './Field';
export * from './Mutation';
export * from './Query';
export * from './Root';
