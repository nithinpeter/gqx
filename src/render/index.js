import { createElement } from '../utils/createElement';
import GQXRenderer from '../reconciler/';

function render(element, path) {
  const container = createElement('ROOT');
  const node = GQXRenderer.createContainer(container);
  GQXRenderer.updateContainer(element, node, null);

  return container.document.render();
}

export default render;
