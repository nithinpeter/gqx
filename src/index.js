import render from './render';

// Aliases for createElement method
const Field = 'FIELD';
const Query = 'QUERY';
const Mutation = 'MUTATION';

export { render, Field, Query, Mutation };
