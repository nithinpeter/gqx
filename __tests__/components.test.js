import React, { Component } from 'react';
import { Query, Mutation, Field, render } from '../src';

it('should render query', () => {
  const query = render(
    <Query name="DogDetailsQuery">
      <Field name="dogName" />
      <Field name="human" alias="hooman">
        <Field name="address">
          <Field name="addressLine1" />
          <Field name="state" />
        </Field>
      </Field>
    </Query>
  );
  expect(query).toMatchSnapshot();
});

it('should render mutation', () => {
  const mutation = render(
    <Mutation name="DogDetailsQuery">
      <Field name="dogName" />
      <Field name="human" alias="hooman">
        <Field name="address">
          <Field name="addressLine1" />
          <Field name="state" />
        </Field>
      </Field>
    </Mutation>
  );
  expect(mutation).toMatchSnapshot();
});
