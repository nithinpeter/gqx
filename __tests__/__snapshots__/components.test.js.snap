// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render mutation 1`] = `"mutation DogDetailsQuery { dogName,hooman: human  { address  { addressLine1,state } } }"`;

exports[`should render query 1`] = `"query DogDetailsQuery { dogName,hooman: human  { address  { addressLine1,state } } }"`;
